app/controllers/data_explorer/query_controller.rb

```diff
@@ -0,0 +1,236 @@
+class DataExplorer::QueryController < ::ApplicationController
+  requires_plugin DataExplorer.plugin_name
+
+  before_action :check_enabled
```

Comment:
`requires_plugin` is already setting a `before_action` to check that the plugin has not been disabled.